src/heitech.zer0mqXt.core/Main/Zer0Mq.cs

```diff
@@ -28,6 +28,14 @@ public IZer0MqBuilder SetSerializer(ISerializerAdapter adapter)
             return this;
         }
 
+        public IZer0MqBuilder UseNewtonsoftJson()
+        {
+            var adapter = new InternalAdapter();
```

Comment:
Do we need to instantiate a new Adapter here? Could we perhaps just directly `_serializer.SetToNewtonsoft();` on the already existing instance?

Reply:
the member _serializer is of Type ISerializerAdapter which does not offer this method. And a downcast is just bad style here. 

So it seems to me, that this is the best solution without changing too much of the code or introducing unecessary indirections.

Reply:
Ah, I see.

Reply:
nvmnd I think I found another sln which might make both of us happy, and I could at the same time utilize the ??= operator :>

Reply:
done

Reply:
Very happy 😉. I love the change indeed.